python/GafferSceneUITest/ParameterInspectorTest.py

```diff
@@ -878,6 +878,125 @@ def testShaderAndParameter( self ) :
+			source = s["addA"]["parameters"]["a"],
+			sourceType = GafferSceneUI.Private.Inspector.Result.SourceType.Other,
+			editable = True,
+			edit = s["addA"]["parameters"]["a"]
+		)
+
+	def testTwoConnectionsToSameNode( self ) :
+
+		s = Gaffer.ScriptNode()
+
+		s["add"] = GafferScene.Shader()
+		s["add"]["parameters"]["a"] = Gaffer.Color3fPlug()
+		s["add"]["parameters"]["b"] = Gaffer.FloatPlug()
+		s["add"]["out"] = Gaffer.Color3fPlug( direction = Gaffer.Plug.Direction.Out )
+
+		s["light"] = GafferSceneTest.TestLight()
+		s["light"]["parameters"]["intensity"].setInput( s["add"]["out"] )
+		s["light"]["parameters"]["intensity2"] = Gaffer.Color3fPlug()
+		s["light"]["parameters"]["intensity2"].setInput( s["add"]["out"] )
+
+		self.__assertExpectedResult(
+			self.__inspect( s["light"]["out"], "/light", "a", shader = "add" ),
+			source = s["add"]["parameters"]["a"],
+			sourceType = GafferSceneUI.Private.Inspector.Result.SourceType.Other,
+			editable = True,
+			edit = s["add"]["parameters"]["a"]
+		)
+
+		self.__assertExpectedResult(
+			self.__inspect( s["light"]["out"], "/light", "b", shader = "add" ),
+			source = s["add"]["parameters"]["b"],
+			sourceType = GafferSceneUI.Private.Inspector.Result.SourceType.Other,
+			editable = True,
+			edit = s["add"]["parameters"]["b"]
+		)
+
+	def testIdenticalShaderNames( self ) :
+
+		s = Gaffer.ScriptNode()
+
+		s["box"] = Gaffer.Box()
+
+		s["box"]["add"] = GafferScene.Shader()
+		s["box"]["add"]["parameters"]["a"] = Gaffer.Color3fPlug()
+		s["box"]["add"]["out"] = Gaffer.Color3fPlug( direction = Gaffer.Plug.Direction.Out )
+
+		s["box"]["outColor"] = Gaffer.BoxOut()
+		s["box"]["outColor"].setup( s["box"]["add"]["out"] )
+		s["box"]["outColor"]["in"].setInput( s["box"]["add"]["out"] )
+
+		s["add"] = GafferScene.Shader()
+		s["add"]["parameters"]["a"] = Gaffer.FloatPlug()
+		s["add"]["out"] = Gaffer.FloatPlug( direction = Gaffer.Plug.Direction.Out )
+
+		s["light"] = GafferSceneTest.TestLight()
+		s["light"]["parameters"]["intensity"].setInput( s["box"]["out"] )
+		s["light"]["parameters"]["exposure"].setInput( s["add"]["out"] )
+
+		self.assertEqual( s["box"]["add"].getName(), s["add"].getName() )
+
+		# `intensity` comes first in the parameter list and will be added to the network first
+		self.__assertExpectedResult(
+			self.__inspect( s["light"]["out"], "/light", "a", shader = "add" ),
+			source = s["box"]["add"]["parameters"]["a"],
+			sourceType = GafferSceneUI.Private.Inspector.Result.SourceType.Other,
+			editable = True,
+			edit = s["box"]["add"]["parameters"]["a"]
+		)
+
+		self.__assertExpectedResult(
+			self.__inspect( s["light"]["out"], "/light", "a", shader = "add1" ),
+			source = s["add"]["parameters"]["a"],
+			sourceType = GafferSceneUI.Private.Inspector.Result.SourceType.Other,
+			editable = True,
+			edit = s["add"]["parameters"]["a"]
+		)
+
+	# \todo :
+	# - Test component connections
+	# - Test splines
+	# - Test disabled node passthrough
+	# - Detect cycles
```

Comment:
I think we can forego more exhaustive testing now that we know the internal implementation is using NetworkBuilder.